research-software-directory/RSD-as-a-service
# Auto fill in keywords

# Auto fill keywords

Now same as the contributors we can get the keywords from the DOI.
See #201 and https://github.com/research-software-directory/Helmholtz/issues/9

@dmijatovic you may have to improve the UI again (you can add the commits in this PR if you want :wink:), for example the page scrolls to top after keywords were loaded which is not ideal, but I don't know how to prevent this nicely since we would need to add anchors to the headings (?)

I want also to do the same for the license, since we can get it from DataCite too, but this will be another PR. But those three things (contributors, keywords, license) are all we can get from a DOI automatically for now, I think...

Changes proposed in this pull request:

* import the keywords from the DataCite API if a DOI was given

How to test:

* create a new software entry and enter a DOI which has keywords in the data (for example: 10.6084/m9.figshare.90828 or 10.5880/riesgos.2021.003)
* click on "Import Keywords", keywords should be added, but not saved yet, no duplicate keywords are added
* a DOI without keyword information does not add any keywords (example: 10.5281/zenodo.2585848)

PR Checklist:
